ol3-viewer/src/ome/ol3/utils/Conversion.js

```diff
@@ -527,84 +482,81 @@ ome.ol3.utils.Conversion.LOOKUP = {
-	 		'http://www.openmicroscopy.org/Schemas/OME/2016-06#Label';
+    var presentStyle = feature.getStyle();
+    if (typeof(presentStyle) === 'function') presentStyle = presentStyle(1);
+    if (ome.ol3.utils.Misc.isArray(presentStyle)) presentStyle = presentStyle[0];
+    if (!(presentStyle instanceof ol.style.Style)) return;
+
+    var isLabel =
+        jsonObject['@type'] ===
+            'http://www.openmicroscopy.org/Schemas/OME/2016-06#Label';
     var presentFillColor =
         isLabel && presentStyle.getText() && presentStyle.getText().getFill() ?
             presentStyle.getText().getFill().getColor() :
                 presentStyle.getFill() ? presentStyle.getFill().getColor() : null;
-	if (presentFillColor)
-		jsonObject['FillColor'] =
-			ome.ol3.utils.Conversion.convertColorToSignedInteger(presentFillColor);
-
-	var presentStrokeStyle =
-		isLabel && presentStyle.getText() && presentStyle.getText().getFill() ?
-			presentStyle.getText().getFill().getColor() :
-				((typeof(feature['oldStrokeStyle']) === 'object' &&
-                    feature['oldStrokeStyle'] !== null &&
-					typeof(feature['oldStrokeStyle']['color']) !== 'undefined') ?
-						feature['oldStrokeStyle']['color'] : null);
-
-	if (presentStrokeStyle) { // STROKE
-			jsonObject['StrokeColor'] =
-				ome.ol3.utils.Conversion.convertColorToSignedInteger(presentStrokeStyle);
-	}
-	var presentStrokeWidth =
-		(typeof(feature['oldStrokeStyle']) === 'object' &&
-			feature['oldStrokeStyle'] !== null &&
-			typeof(feature['oldStrokeStyle']['width']) === 'number') ?
-				feature['oldStrokeStyle']['width'] : 1;
-
-	jsonObject['StrokeWidth'] = {
-		'@type': 'TBD#LengthI',
-		'Unit': 'PIXEL',
-		'Symbol': 'px',
-		'Value':
-			isLabel && presentStyle.getText() && presentStyle.getText().getStroke() ?
-			presentStyle.getText().getStroke().getWidth() : presentStrokeWidth
-	};
+    if (presentFillColor)
+        jsonObject['FillColor'] =
+            ome.ol3.utils.Conversion.convertColorToSignedInteger(presentFillColor);
+
+    var presentStrokeStyle =
+        isLabel && presentStyle.getText() && presentStyle.getText().getFill() ?
+            presentStyle.getText().getFill().getColor() :
+                ((typeof(feature['oldStrokeStyle']) === 'object' &&
+                feature['oldStrokeStyle'] !== null &&
+                typeof(feature['oldStrokeStyle']['color']) !== 'undefined') ?
+                    feature['oldStrokeStyle']['color'] : null);
+
+    if (presentStrokeStyle) { // STROKE
+        jsonObject['StrokeColor'] =
+            ome.ol3.utils.Conversion.convertColorToSignedInteger(
+                presentStrokeStyle);
+    }
+    var presentStrokeWidth =
+        (typeof(feature['oldStrokeStyle']) === 'object' &&
+        feature['oldStrokeStyle'] !== null &&
+        typeof(feature['oldStrokeStyle']['width']) === 'number') ?
+            feature['oldStrokeStyle']['width'] : 1;
+
+    jsonObject['StrokeWidth'] = {
+        '@type': 'TBD#LengthI',
```

Comment:
yes, that's omero marshal. I have found it wants that giving it the stroke width for marshalling ;-)